### PR TITLE
Reset tcp connection earlier

### PR DESCRIPTION
**Pull request by sbuliarca**

### CODE REVIEW

File: ansible/templates/cluster-template.yaml.j2 [hunk 1]
```diff
@@ -630,6 +630,15 @@ worker:
...
           RateLimitBurst=0
 
+#  Tuned the net.ipv4.tcp_retries2 kernel param on the worker nodes, so that stale TCP connections will be closed in ~ 1min.
+#  Since the kubelet holds a long TCP connection to the API server for posting its status, if that connection gets stalled, we'll get a NodeNotReady
+#  The default was 15 which caused the stale connections to be closed in ~ 15 mins.
+#  More on this here:  https://github.com/kubernetes/kubernetes/pull/48670, https://github.com/kubernetes/kubernetes/issues/41916#issuecomment-312428731
+      - path: /etc/sysctl.d/90-tcp-retries.conf
+        permissions: 0644
+        content: |
+          net.ipv4.tcp_retries2=6
```
Comment (efinlay24) [inline ansible/templates/cluster-template.yaml.j2:640]: Does this mean we will need to wait up to 6 minutes before a node's connection is dropped and it reconnects?

Do we need to drop the retry count down lower?

Comment (sbuliarca) [reply ansible/templates/cluster-template.yaml.j2:640]: Not really, so a value of 6 closes the connection in less than 60s as I tested. I wrote it in the comment :)

Comment (efinlay24) [reply ansible/templates/cluster-template.yaml.j2:640]: Cool. 👍

**Review by efinlay24 — APPROVED** [review]

**Comment by bobbytables** [comment]
Hey we found this while debugging an issue (which it seems like you had the same one). I'm curious if your nodes go into NotReady state but just for a shorter amount of time?

**Comment by sbuliarca** [comment]
@bobbytables We manage to reduce the appearances of NodeNotReady with this workaround + this one https://github.com/Financial-Times/content-k8s-provisioner/pull/187/files#diff-c5da90491d4c1629023320b698a91216R642 , as the connection between the nodes and the API server is reestablished in the 1 minute time window, but we still get it sometimes, when it lasts the same 15 minutes.

**Pull request merged**